### PR TITLE
Add two channel videos

### PR DESCRIPTION
**Pull request by fanimator12**

### CODE REVIEW

File: ui/src/routes/lecture.tsx [hunk 1]
```diff
@@ -24,21 +24,33 @@ function filenameFromUrl(url: string) {
...
-      setLecture({});
+      return error;
     }
   };
 
   useEffect(() => {
-    fetchLecture();
+    (async () => {
+      await fetchLecture();
+    })();
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:53]: it's not a good practice to have async here

File: ui/src/routes/lecture.tsx [hunk 2]
```diff
@@ -24,21 +24,33 @@ function filenameFromUrl(url: string) {
...
       setLoading(true);
       const results = await getApiRoot().get(`/lectures/${lectureId}/`);
-      setLoading(false);
       setLecture(results.data);
+
+      const year = await results.data.date.slice(0, 4);
+      setYear(year);
+
+      const indicoId = await results.data.event_details.split("/")[4];
+      setIndicoId(indicoId);
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:41]: - why async? this operations are not returning a promise.
- you don't have to use state it's just a lookup 
- Indico will not work

File: ui/src/routes/lecture.tsx [hunk 3]
```diff
@@ -24,21 +24,33 @@ function filenameFromUrl(url: string) {
...
+
+      setLoading(false);
+      return;
     } catch (error) {
-      setLecture({});
+      return error;
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:46]: why returning an error? the whole page will fail

File: harvest/harvest/spiders/cds_spider.py [hunk 4]
```diff
@@ -259,11 +259,18 @@ def parse_item(self, selector, original=None):
 
         record["files"] = [_file["a"] for _file in data.get("FFT__", [])]
 
+        file_types = selector.xpath(
+            './/datafield[@tag=856]/subfield[@code="x"]/text()'
+        ).get()
```
Comment (drjova) [inline harvest/harvest/spiders/cds_spider.py:264]:
```suggestion
        ).getall()
```

File: ui/src/routes/lecture.tsx (hunk 1 above)
Comment (fanimator12) [reply ui/src/routes/lecture.tsx:53]: Noted.

File: ui/src/routes/lecture.tsx [hunk 5]
```diff
@@ -24,14 +24,33 @@ function filenameFromUrl(url: string) {
...
+  let year = null;
+  let indicoId = null;
+
+  if (isSlide && lecture.date) {
+    year = lecture.date.slice(0, 4);
+  }
+
+  if (isSlide && lecture.event_details) {
+    indicoId = lecture.event_details.split("/")[4];
+    console.log(indicoId);
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:40]: you don't need the console.log

File: ui/src/routes/lecture.tsx [hunk 6]
```diff
@@ -24,14 +24,33 @@ function filenameFromUrl(url: string) {
...
   let { lectureId } = useParams();
 
   const fetchLecture = async () => {
     try {
       setLoading(true);
       const results = await getApiRoot().get(`/lectures/${lectureId}/`);
-      setLoading(false);
       setLecture(results.data);
+      setLoading(false);
+      return;
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:53]: you don't need the `return`

**Pull request merged**